Sources/Rendering/Core/Mapper/index.js

```diff
@@ -545,6 +552,70 @@ export function extend(publicAPI, model, initialValues = {}) {
     'scalarRange',
   ], 2);
 
+  model.viewSpecificProperties = {
+    OpenGL: {
+      ShaderReplacements: [],
+      VertexShaderCode: '',
+      FragmentShaderCode: '',
+      GeometryShaderCode: '',
+    },
+
+    addShaderReplacement(_shaderType, _originalValue, _replaceFirst, _replacementValue, _replaceAll) {
+      this.OpenGL.ShaderReplacements.push({
+        shaderType: _shaderType,
+        originalValue: _originalValue,
+        replaceFirst: _replaceFirst,
+        replacementValue: _replacementValue,
+        replaceAll: _replaceAll,
+      });
+    },
+
```

Comment:
I did, but you could probably just make it  a function in the example and not stick it in the properties. That is probably easier/cleaner.